blog/writing-reusable-julia-code.md

```diff
@@ -0,0 +1,241 @@
+    
+    ```
+    (MyPackage) pkg> st
+    Project MyPackage v0.1.0
+    Status `C:\Users\MyUserName\Documents\DSCoV-Reusable-Julia\MyPackage\Project.toml` (empty project)
+
+    (MyPackage) pkg> 
+    ```
+
+    In the error message above Julia is suggesting that we have an undocumented/uninstalled/broken dependency. Here we just need to add it to our project. Let's do it! 
+    ```
+    (MyPackage) pkg> add Example
+       Resolving package versions...
+        Updating `C:\Users\MyUserName\Documents\Reusable-code-Julia\MyPackage\Project.toml`
+      [7876af07] + Example v0.5.3
+        Updating `C:\Users\MyUserName\Documents\Reusable-code-Julia\MyPackage\Manifest.toml`
+      [7876af07] + Example v0.5.3
+    Precompiling project...
+      1 dependency successfully precompiled in 1 seconds. 1 already precompiled.
+    ```
+    ```
+    (MyPackage) pkg> st # again seeing is believing
+    Project MyPackage v0.1.0
+    Status `C:\Users\MyUserName\Documents\Reusable-code-Julia\MyPackage\Project.toml`
+    [7876af07] Example v0.5.3
+    ```
+
+    Now that the `Example` dep is documented and compiled let's try testing our package again! 
+    ```
+    julia> using MyPackage
+
+    julia> fact()
+    FYI, 2 + 5 = 7
+
+    julia>
+    ```
+Success! Our feature works and we are ready (sort of) to share our package with the world!
+
+### Recommended Configuration
+
+Before sharing your code with the world, especially larger code bases, it's a good idea to have our code conform to a standard style. There is tooling for this available and there are many associated benefits:
+
+1. **Consistency**: It ensures that your code has a consistent style, which makes it easier to read and understand.
+
+2. **Saves Time**: You don't have to spend time manually formatting your code or arguing about coding styles in code reviews (when working with other people).
+
+3. **Prevents Bugs**: Some formatters can catch and fix minor errors, such as missing semicolons, unused variables, etc.
+
+4. **Integration with IDEs**: Most modern IDEs support automatic formatting on save, which makes it effortless to keep your code well-formatted.
+
+5. **Focus on Logic**: With automatic formatting, you can focus on the logic of your code, rather than its appearance.
+
+
+Here is a suggested configuration for VS Code with the Julia extension.
+1.  Add a `.JuliaFormatter.toml` file with the following 
+    ```toml
+    style = "blue"
+    ```
+    Read about the *blue style* [here](https://blue.readthedocs.io/en/latest/).
+
+2.  Add a `.gitignore` file 
+    ```
+    .JuliaFormatter.toml
+    ```
+The `.gitignore` file will prevent the configuration file just created from getting *pushed* to our remote repository. Here we have added the `.JuliaFormatter.toml` configuration file we created earlier.  
+
+ 
+## Publish your package!
```

Comment:
So in this section you're pushing the package into a new git repository. Is it just the package or is it all of the code?

Also, it might be nice to instead make a template repo, or to make a git repo at the beginning

Reply:
I'm wondering if it would make more sense to create this repo in the beginning of this project than at the end of it all.

Reply:
I think it doesn't matter too much as the focus is not on using git. I kind of prefer to do it at the end.